code/segmentedtickerplant/datastripe.q

```diff
@@ -0,0 +1,74 @@
+//Defining a function that loads configurable .csv files into memory so relevant filters can be applied to data.
+//calls segmenting.csv which customer edits as to how many segments there will be.
+//calls filtermap.csv which customer edits as to what filters should be applied to each segment.
+
+//.ds.segmentconfig/segmentfiltermap are variables which can take different csv files. These files can be chosen on in process.csv.
+//error trap these .ds variables incase this file is loaded stand alone.
+
+.ds.segmentconfig:@[value;`.ds.segmentconfig;`segmenting.csv];
+.ds.filtermap:@[value;`.ds.filtermap;`filtermap.csv];
+
+//if statements check segmenting.csv and filtermap.csv exist. If not, process exited and message sent to error logs.
+//configload transfoms segmenting.csv into table format so it can be accessed.
+//configload transforms filtermap.csv into table format then into a mapping of wcRef to filter which can be accessed and applied to data.
+//checks csv files load correctly. If not, process exited and message sent to error logs.
+
+configload:{
+     if[()~key hsym scpath:first .proc.getconfigfile[string .ds.segmentconfig];.lg.e[`init;"The following file can not be found: ",string .ds.segmentconfig]];
+     if[()~key hsym fmpath:first .proc.getconfigfile[string .ds.filtermap];.lg.e[`init;"The following file can not be found: ",string .ds.filtermap]];
+     @[{.stpps.segmentconfig:("SIS";enlist",")0: hsym x};scpath;{.lg.e[`init;"Failure in loading ",string y]}[;scpath]];
```

Comment:
Is the logic here actually
$[if no file
   log no file;
   attempt to process file]

Seems we are doing checks first in an order, then processing in an order.  I think would be better flow to treat each config load in order.